models/plan.py

```diff
@@ -0,0 +1,36 @@
+from collections import Iterable
+from datetime import date
+from enum import Enum
+
+from models.camp import Camp
+
+
+class Plan:
+    class EmergencyType(Enum):
+        EARTHQUAKE = 'earthquake'
+        FIRE = 'fire'
+        TSUNAMI = 'tsunami'
+        STORM = 'storm'
+
+    def __init__(self,
+                 name: str,
+                 emergency_type: EmergencyType,
+                 description: str,
+                 geographical_area: str,
+                 start_date: date,
+                 camps: Iterable[Camp] = None):
+        self.name = name
+        self.emergency_type = emergency_type
+        self.description = description
+        self.geographical_area = geographical_area
+        self.start_date = start_date
+        self.camps: set[Camp] = set(camps) if camps else set()
+
```

Comment:
Changes based on feedback: 

Ensuring that start_date is valid by including the following in class Plan:
```
    def check_date(self, start_date):
        try:
            startDate = datetime.datetime.strptime(start_date, "%d/%m/%Y")
            present = datetime.datetime.now()
            if startDate < present:
                raise PastStartDate(start_date)
            else:
                return startDate
        except:
            raise ValidStartDateString(start_date)
```

And the exception classes: 

```

# Exception handling for start date:
class PastStartDate(Exception):
    """Exception raised when start date entered is in the past.

    Attributes:
        start_date - input start_date which caused the error
        message - explanation of the error
    """

    def __init__(self, start, message="Start date is in the past. Please enter a valid start date."):
        self.start = start
        self.message = message
        super().__init__(self.message)


class ValidStartDateString(Exception):
    """Exception raised when start date entered is not a valid string input.

    Attributes:
        start_date - input start_date which caused the error
        message - explanation of the error
    """

    def __init__(self, start, message="Invalid start date entered. Please enter a valid string of the form: dd/mm/yy."):
        self.start = start
        self.message = message
        super().__init__(self.message)

```

Adding tests at the end to check if the exception handling was correct:

```
# Testing whether exception handling works:
# Pass case:
planPass = Plan('My Plan',
                emergency_type=Plan.EmergencyType.EARTHQUAKE,
                description='Test emergency plan',
                geographical_area='',
                start_date="09/11/2022")

print(planPass.check_date(start_date="09/11/2022"))

# PastStartDate error:
planFail = Plan('My Plan',
                emergency_type=Plan.EmergencyType.EARTHQUAKE,
                description='Test emergency plan',
                geographical_area='',
                start_date="09/11/2020")

print(planFail.check_date(start_date="09/11/2020"))

#ValidStartDateString error:
planFail2 = Plan('My Plan',
                emergency_type=Plan.EmergencyType.EARTHQUAKE,
                description='Test emergency plan',
                geographical_area='',
                start_date="random string")

```